core/src/validator.rs

```diff
@@ -343,6 +344,26 @@ impl Validator {
                 .set_entrypoint(entrypoint_info.clone());
         }
 
+        if let Some(ref snapshot_config) = config.snapshot_config {
```

Comment:
I'll pull this hunk, as well as the validator.rs changes in this PR into my reworked version of https://github.com/solana-labs/solana/pull/8364/files later today.   I'm thinking that I'll modify validator.rs to not both even trying to download a snapshot that's not advertised by the trusted validator set.